src/main/java/com/redhat/devtools/intellij/knative/tree/KnRevisionDescriptor.java

```diff
@@ -48,27 +53,35 @@ protected void update(@NotNull PresentationData presentation) {
                 break;
             }
         }
-        if (Strings.isNullOrEmpty(errorMessage)) {
-            presentation.setTooltip("Revision: " + element.getName());
-            presentation.addText(element.getName() + " ", SimpleTextAttributes.REGULAR_ATTRIBUTES);
-        } else {
-            presentation.setTooltip(errorMessage.length() >= 100 ? errorMessage.substring(0, 100) + "..." : errorMessage);
-            presentation.addText(element.getName() + " ", SimpleTextAttributes.ERROR_ATTRIBUTES);
-        }
+
+        StringBuilder tag = new StringBuilder();
+        int percent = 0;
         if (status != null && status.getTraffic() != null) {
-            StringBuilder tag = new StringBuilder();
-            int percent = 0;
             for (ServiceTraffic st : status.getTraffic()) {
                 if (st.getRevisionName().equals(element.getName())) {
                     percent += Math.max(st.getPercent(), 0);
                     tag.append(st.getLatestRevision() ? "latest" : "");
-                    tag.append(st.getTag());
+                    if (st.getTag() != null) {
+                        tag.append(" ").append(st.getTag());
+                    }
                 }
             }
-            presentation.addText(percent + "%", SimpleTextAttributes.GRAYED_ITALIC_ATTRIBUTES);
-            presentation.addText(" " + tag.toString(), SimpleTextAttributes.GRAY_ATTRIBUTES);
 
+        }
 
+        if (Strings.isNullOrEmpty(errorMessage)) {
+            presentation.setTooltip("Revision: " + element.getName());
+            presentation.addText(element.getName(), percent != 0 ? SimpleTextAttributes.REGULAR_ATTRIBUTES : WARNING_ATTRIBUTES);
+        } else {
+            presentation.setTooltip(errorMessage.length() >= 100 ? errorMessage.substring(0, 100) + "..." : errorMessage);
```

Comment:
I think this kind of logic should be extracted in a separate class utility as it's likely to be used elsewhere